locking/lockable.go

```diff
@@ -0,0 +1,210 @@
+package locking
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/git-lfs/git-lfs/config"
+	"github.com/git-lfs/git-lfs/errors"
+	"github.com/git-lfs/git-lfs/filepathfilter"
+	"github.com/git-lfs/git-lfs/git"
+	"github.com/git-lfs/git-lfs/tools"
+)
+
+// GetLockablePatterns returns a list of patterns in .gitattributes which are
+// marked as lockable
+func (c *Client) GetLockablePatterns() []string {
+	c.ensureLockablesLoaded()
+	return c.lockablePatterns
+}
+
+// getLockableFilter returns the internal filter used to check if a file is lockable
+func (c *Client) getLockableFilter() *filepathfilter.Filter {
+	c.ensureLockablesLoaded()
+	return c.lockableFilter
+}
+
+func (c *Client) ensureLockablesLoaded() {
+	c.lockableMutex.Lock()
+	defer c.lockableMutex.Unlock()
+
+	// Only load once
+	if c.lockablePatterns == nil {
+		c.refreshLockablePatterns()
+	}
+}
+
+// Internal function to repopulate lockable patterns
+// You must have locked the c.lockableMutex in the caller
+func (c *Client) refreshLockablePatterns() {
+	// Always make non-nil even if empty
+	c.lockablePatterns = make([]string, 0, 10)
+
+	paths := git.GetAttributePaths(config.LocalWorkingDir, config.LocalGitDir)
```

Comment:
OK; even after doing this it will still require the `config` package, until the per-package config comes in, but at least it puts it in one place ready for that I guess.